tests/e2e/systemtest/mon/test_restore_ceph_mon_quorum.py

```diff
@@ -0,0 +1,135 @@
+import logging
+
+import pytest
+
+from ocs_ci.framework.testlib import E2ETest, ignore_leftovers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.pod import wait_for_storage_pods, get_ceph_tools_pod
+from ocs_ci.helpers.helpers import (
+    mon_quorom_lost,
+    recover_mon_quorum,
+    modify_deployment_replica_count,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+
+log = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+class TestRestoreCephMonQuorum(E2ETest):
+    """
+    The objective of this test case is to verify that
+    mons can be brought to quorum successfully by
+    following the steps mentioned in
+    https://access.redhat.com/solutions/5898541, when
+    mon(s) are out of quorum on a cluster
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def base_set_up(self):
+        """
+        Preconditions:
+
+        a) Have a cluster with FIPS and Hugepages enabled
+        b) Create some resources s3 objects, buckets and write data's to it.
+        c) Create RGW kafka notification & see the objects are notified to kafka
+        d) Perform mcg bucket replication (bidirectional) and see the objects are synced.
+        e) Perform noobaa caching
+        f) there are snapshots and clones for few PVCs
+        g) Background IOs (fio pods) running
+        """
+        # ToDo:Validate FIPS, hugepages enabled in cluster
+
+        # ToDo: Create objects and write data's to it
+
+        # ToDo: Create RGW kafka notification
+
+        # ToDo: Perform mcg bucket replication
+
+        # Todo: Noobaa caching
+
+        # ToDo: Create PVC, snapshot and restore into new PVC, clone PVC
+
+        # ToDo: Run background IOs
+
+    @pytest.fixture(autouse=True)
+    def base_teardown(self, request):
+        def finalizer():
+            op_obj = OCP(
+                kind=constants.DEPLOYMENT,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+            op_obj = op_obj.get(resource_name="rook-ceph-operator")
```

Comment:
Do you think we need to create a teardown fixture for rook ceph operator pod  teardown? This can help in our system tests, wdyt?

Reply:
Agreed, not sure will be required in our system tests case. I will move it later if required this teardown function, is that okay?